cumulus/polkadot-omni-node/lib/src/cli.rs

```diff
@@ -128,8 +128,9 @@ pub struct Cli<Config: CliConfig> {
 	///
 	/// This is a dev option, and it won't result in starting or connecting to a parachain network.
 	/// The resulting node will work on its own, running the wasm blob and artificially producing
-	/// a block each `dev_block_time` ms, as if it was part of a parachain.
-	#[arg(long)]
+	/// a block each `dev_block_time` ms, as if it was part of a parachain. Default to 3000ms if
+	/// not set when using `--dev` flag.
+	#[arg(long, requires = "dev")]
```

Comment:
Hmm why does this require `--dev`? If I just specify `--dev-block-time 500` it could just work like before no? Basically `--dev` just being a shortcut.

Reply:
> Maybe I am missing something, but `--dev` can keep doing this stuff. But I could still be able to specify `--dev-block-time` on its own like before.

Yes, it can still do its stuff, but that stuff wouldn't be activated if we're not setting the `--dev` flag, right? I mean, we can also set it artificially like below:
```rust
cli.run.base.shared_params.dev |= cli.dev_block_time.is_some();
```

Is this what you think about?

Reply:
Yes exactly. I mean this is what I meant by "old behaviour". The check was changed in this PR, before we were checking for dev_block_time already.

So basically my idea is:
- `--dev`: Activates manual seal with default 3000ms block time. And all the other flags you mentioned, basically easy dev setup.
- `--dev-block-time 1000`: Just manual seal with 1000ms, no extra hidden flags activated.
- `--dev --dev-block-time 1000`: The easy dev setup but with 1000ms block times.

Let me know if you think it makes sense.

Reply:
Yes, I am ok with this. Changed it back here: [06bbd5e](https://github.com/paritytech/polkadot-sdk/pull/6646/commits/06bbd5e6c54a657dc989db2d5032764b32475ccf), and adjusted a few more things to reflect that `--dev` isn't mandatory.